src/conversions.jl

```diff
@@ -705,6 +712,61 @@ function convert_attribute(ls::Symbol, ::key"linestyle")
     end
 end
 
+function diff_pattern(ls_str::AbstractString, gaps = :normal)
+    dot = 1
+    dash = 3
+    check_pattern(ls_str)
+    
+    dot_gap, dash_gap = convert_gaps(gaps)
+    
+    pattern = Float64[]
+    for i in 1:length(ls_str)
+        curr_char = ls_str[i]
+        next_char = i == lastindex(ls_str) ? ls_str[begin] : ls_str[i+1]
+        # push dash or dot
+        if curr_char == '-'
+            push!(pattern, dash)
+        else
+            push!(pattern, dot)
+        end
+        # push the gap (use dot_gap only between two dots)
+        if (curr_char == '.') && (next_char == '.')
+            push!(pattern, dot_gap)
+        else
+            push!(pattern, dash_gap)
+        end
+    end
+    pattern
+end
+
+"Checks if the linestyle format provided as a string contains only dashes and dots"
+function check_pattern(ls_str)
+    for i in 1:length(ls_str)
```

Comment:
this would fail with unicode characters, because you can't index into strings like that. minor point, but maybe just replace with this small regex:

```julia
isnothing(match(r"^[\.\-]+$", ls_str)) && error...
```

Reply:
Thanks, I'll use your regex, even though I've no clue how to read it ;-)

Reply:
it means:
`^` start of string
`[\.\-]` a set containing the . and - character
`+` at least one character from that set
`$` end of string

but I just checked and in this set . and - don't need to be escaped, you can just use `r"^[.-]+$"` then

Reply:
Thanks, TIL :-)